packages/fx-core/src/plugins/resource/frontend/configs.ts

```diff
@@ -109,6 +80,67 @@ export class FrontendConfig {
       });
   }
 
+  static getStorageName(ctx: PluginContext, getFromArmOutput = false): string {
+    let result = isArmSupportEnabled()
+      ? getStorageAccountNameFromResourceId(
+          FrontendConfig.getStorageResourceId(ctx, getFromArmOutput)
+        )
+      : ctx.config.getString(FrontendConfigInfo.StorageName);
+    if (!result) {
+      const resourceNameSuffix = FrontendConfig.getConfig<string>(
+        DependentPluginInfo.ResourceNameSuffix,
+        ctx.envInfo.profile.get(DependentPluginInfo.SolutionPluginName)
+      );
+      result = Utils.generateStorageAccountName(
+        ctx.projectSettings!.appName,
+        resourceNameSuffix,
+        Constants.FrontendSuffix
+      );
+    }
+    if (!RegularExpr.FrontendStorageNamePattern.test(result)) {
+      throw new InvalidStorageNameError();
+    }
+    return result;
+  }
+
+  static getStorageResourceId(ctx: PluginContext, getFromArmOutput = false): string {
+    const result = getFromArmOutput
+      ? getArmOutput(ctx, ArmOutput.FrontendStorageResourceId)
+      : ctx.config.getString(FrontendConfigInfo.StorageResourceId);
+    if (!result) {
+      throw new Error("Failed to get storaget accounts resource id");
```

Comment:
This error eventually be a `Unhandled Error`. better to throw `InvalidConfigError("resourceId")`. 
Since arm output is not config. Maybe you can define a new error type.

Reply:
Fixed.